packages/railtracks/src/railtracks/vector_stores/chroma.py

```
@@ -0,0 +1,227 @@
+from __future__ import annotations
+from typing import TYPE_CHECKING, List, Optional, Callable
+from copy import deepcopy
+from .vector_store import (
+    VectorStore,
+    Chunk,
+    SearchResponse,
+    FetchResponse,
+    FetchResult,
+    SearchResult,
+    Document,
+    MetadataKeys,
+)
+from uuid import uuid4
+import numpy as np
+
+if TYPE_CHECKING:
+    from chromadb.base_types import Where, WhereDocument
+    from chromadb.api.types import Include
+
+CONTENT = MetadataKeys.CONTENT.value
+
+
+class ChromaVectorStore(VectorStore):
+    """ChromaDB implementation of VectorStore."""
+
+    @classmethod
+    def class_init(cls, path, host, port):
+        if not hasattr(cls, "_chroma"):
+            # First object initialized so import Chroma and connect create temporary, persistent or http client. Don't currently support chroma cloud client
+            try:
+                import chromadb
+
+                if path:
+                    cls._chroma = chromadb.PersistentClient(path=path)
+                elif host and port:
+                    cls._chroma = chromadb.HttpClient(
+                        host=host, port=port
+                    )  # Currently this does not provide the extra asycn features that comes with this
+                else:
+                    cls._chroma = chromadb.EphemeralClient()
+            except ImportError:
+                raise ImportError(
+                    "Chroma package is not installed. Please install railtracks[chroma]."
+                )
+
+    def __init__(
```

add some quick fail fast type stuff to handle when the user provides an unknown sequence of parameters. 

i.e. path and port, nothing but host, etc.